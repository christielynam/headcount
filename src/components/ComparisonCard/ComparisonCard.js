import React from 'react'
import styled from 'styled-components'

const ComparisonCardStyles = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  margin: 0 10px 0 10px;
  p {
    text-align: center
  }
`

const ComparisonCard = ({ activeCount, activeCards, findAverage, compareAverages }) => {
  const comparison = () => {
    if(activeCount === 2) {
      return compareAverages(activeCards[0].location, activeCards[1].location)
    }
  }
  const average = comparison()
  return (
    activeCount === 2 && 
    <ComparisonCardStyles>
      <div className='district'>
        <h4>{activeCards[0].location}</h4>
        <p>{findAverage(activeCards[0].location)}</p>
      </div>
      <div className='compare'>-----{average.compared}-----</div>
      <div className='district'>
        <h4>{activeCards[1].location}</h4>
        <p>{findAverage(activeCards[1].location)}</p>
      </div>
    </ComparisonCardStyles>
  )
}

export default ComparisonCard