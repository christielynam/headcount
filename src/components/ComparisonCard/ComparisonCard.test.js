import React from 'react';
import ComparisonCard from './ComparisonCard';
import { shallow } from 'enzyme'

describe('ComparisonCard', () => {
  let wrapper
  let mockCount
  let mockActiveCards
  let mockFindAverage
  let mockCompareAverages
  
  beforeEach(() => {
    mockCompareAverages = jest.fn()
    mockFindAverage = jest.fn()

    wrapper = shallow(<ComparisonCard activeCount={mockCount} activeCards={mockActiveCards} compareAverages={mockCompareAverages} findAverage={mockFindAverage} />)
  })
  
  it('should match the snapshot', () => {
    mockActiveCards = []
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should match the snapshot', () => {
    mockCount = 2
    mockActiveCards = [{location: 'ACADEMY 20', active: false, stats: {2004: '0.302', 2005: '0.267', 2006: '0.354', 2007: '0.392', 2008: '0.385', 2009: '0.39', 2010: '0.436', 2011: '0.489', 2012: '0.479', 2013: '0.488', 2014: '0.49'}}, {location: 'ADAMS COUNTY 14', active: false, stats: {2004: '0.228', 2005: '0.3', 2006: '0.293', 2007: '0.306', 2008: '0.673', 2009: '1', 2010: '1', 2011: '1', 2012: '1', 2013: '0.998', 2014: '1'}}]
    expect(wrapper).toMatchSnapshot()
  })
})