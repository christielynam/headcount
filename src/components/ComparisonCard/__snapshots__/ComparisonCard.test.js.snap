// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonCard should match the snapshot with 1 active district 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    activeCards={
      Array [
        Object {
          "active": true,
          "location": "ACADEMY 20",
          "stats": Object {
            "2004": "0.302",
            "2005": "0.267",
            "2006": "0.354",
            "2007": "0.392",
            "2008": "0.385",
            "2009": "0.39",
            "2010": "0.436",
            "2011": "0.489",
            "2012": "0.479",
            "2013": "0.488",
            "2014": "0.49",
          },
        },
      ]
    }
    activeCount={1}
    compareAverages={[MockFunction]}
    findAverage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ComparisonCard should match the snapshot with 2 active districts 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    activeCards={
      Array [
        Object {
          "active": true,
          "location": "ACADEMY 20",
          "stats": Object {
            "2004": "0.302",
            "2005": "0.267",
            "2006": "0.354",
            "2007": "0.392",
            "2008": "0.385",
            "2009": "0.39",
            "2010": "0.436",
            "2011": "0.489",
            "2012": "0.479",
            "2013": "0.488",
            "2014": "0.49",
          },
        },
        Object {
          "active": true,
          "location": "ADAMS COUNTY 14",
          "stats": Object {
            "2004": "0.228",
            "2005": "0.3",
            "2006": "0.293",
            "2007": "0.306",
            "2008": "0.673",
            "2009": "1",
            "2010": "1",
            "2011": "1",
            "2012": "1",
            "2013": "0.998",
            "2014": "1",
          },
        },
      ]
    }
    activeCount={2}
    compareAverages={
      [MockFunction] {
        "calls": Array [
          Array [
            "ACADEMY 20",
            "ADAMS COUNTY 14",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Object {
              "ACADEMY 20": 0.407,
              "ADAMS COUNTY 14": 0.709,
              "compared": 0.574,
            },
          },
        ],
      }
    }
    findAverage={
      [MockFunction] {
        "calls": Array [
          Array [
            "ACADEMY 20",
          ],
          Array [
            "ADAMS COUNTY 14",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="district"
        >
          <h4>
            ACADEMY 20
          </h4>
          <p />
        </div>,
        <div
          className="compare"
        >
          -----
          0.574
          -----
        </div>,
        <div
          className="district"
        >
          <h4>
            ADAMS COUNTY 14
          </h4>
          <p />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              ACADEMY 20
            </h4>,
            <p />,
          ],
          "className": "district",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ACADEMY 20",
            },
            "ref": null,
            "rendered": "ACADEMY 20",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "-----",
            0.574,
            "-----",
          ],
          "className": "compare",
        },
        "ref": null,
        "rendered": Array [
          "-----",
          0.574,
          "-----",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              ADAMS COUNTY 14
            </h4>,
            <p />,
          ],
          "className": "district",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ADAMS COUNTY 14",
            },
            "ref": null,
            "rendered": "ADAMS COUNTY 14",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  margin: 0 10px 0 10px;
  p {
    text-align: center
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="district"
          >
            <h4>
              ACADEMY 20
            </h4>
            <p />
          </div>,
          <div
            className="compare"
          >
            -----
            0.574
            -----
          </div>,
          <div
            className="district"
          >
            <h4>
              ADAMS COUNTY 14
            </h4>
            <p />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                ACADEMY 20
              </h4>,
              <p />,
            ],
            "className": "district",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ACADEMY 20",
              },
              "ref": null,
              "rendered": "ACADEMY 20",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "-----",
              0.574,
              "-----",
            ],
            "className": "compare",
          },
          "ref": null,
          "rendered": Array [
            "-----",
            0.574,
            "-----",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                ADAMS COUNTY 14
              </h4>,
              <p />,
            ],
            "className": "district",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ADAMS COUNTY 14",
              },
              "ref": null,
              "rendered": "ADAMS COUNTY 14",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  margin: 0 10px 0 10px;
  p {
    text-align: center
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ComparisonCard should match the snapshot with no active districts 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    activeCards={Array []}
    activeCount={0}
    compareAverages={[MockFunction]}
    findAverage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
